### PR TITLE
Bmt27 changing header and footer links & icon issue

### PR DESCRIPTION
**Pull request by hovnaz**

### CODE REVIEW

File: src/main/java/com/mulberry/blackmulberry/BlackMulberryApplication.java [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
-	@Bean
-	public PasswordEncoder passwordEncoder() {
-		return new BCryptPasswordEncoder();
-	}
+        if (byEmail.isEmpty()) {
+            userRepository.save(User.builder()
+                    .name("admin")
+                    .surname("admin")
+                    .email("admin@mail.com")
+                    .password(passwordEncoder().encode("admin"))
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/BlackMulberryApplication.java]: i am not sure that its good idea to have admin credentials with open in code. so lets find another way. for excample you can get encrypted password and admin's data then add it in a sql insert with flyway. but i am sure that there are another ways too. 
if it spend more time , you can create separate ticket and will chenge later

File: src/main/java/com/mulberry/blackmulberry/controller/main/AccountController.java [hunk 2]
```diff
@@ -2,13 +2,14 @@
 
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
 
+@RequestMapping("account")
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/controller/main/AccountController.java]: cool that you add request mapping, but i think 'account' url now is in locale : localhost:8080account, probebly you need add / for its be '/account' -> localhost:8080/account

File: src/main/java/com/mulberry/blackmulberry/controller/main/AuthController.java [hunk 3]
```diff
@@ -2,14 +2,15 @@
 
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
+@RequestMapping("auth")
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/controller/main/AuthController.java]: the same with /

File: src/main/java/com/mulberry/blackmulberry/controller/main/BasketController.java [hunk 4]
```diff
@@ -2,13 +2,15 @@
 
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
+@RequestMapping("basket")
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/controller/main/BasketController.java]: lets check in all controllers

File: src/main/java/com/mulberry/blackmulberry/controller/main/OrdersController.java [hunk 5]
```diff
@@ -6,15 +6,17 @@
...
 @RequiredArgsConstructor
+@RequestMapping("orders")
 public class OrdersController {
     private final ProductRepository productRepository;
 
-    @GetMapping("/orders")
+    @GetMapping("/")
     public String ordersPage(ModelMap modelMap){
         // todo find all products in orders
         List<Product> products = productRepository.findAll();
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/controller/main/OrdersController.java:22]: please dont write logic in the controller and dont use repository here. create productService and invoke logic execution from there

File: src/main/java/com/mulberry/blackmulberry/entity/User.java [hunk 6]
```diff
@@ -16,14 +17,17 @@
...
     private String email;
     private String phone;
-    @Enumerated(value = EnumType.STRING)
-    private Role role;
+    @Enumerated(EnumType.STRING)
+    private UserRole role;
     private String password;
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "create_at", nullable = false, updatable = false)
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/entity/User.java:31]: good job

File: src/main/java/com/mulberry/blackmulberry/exception/DuplicateResourceException.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package com.mulberry.blackmulberry.exception;
+
+public class DuplicateResourceException extends Exception {
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/exception/DuplicateResourceException.java]: is it okay that your exception is checked?

File: src/main/java/com/mulberry/blackmulberry/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    public void save(User user) throws DuplicateResourceException {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()){
+            throw new DuplicateResourceException("User already exists!");
+        }
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        userRepository.save(user);
+    }
+
+    public Optional<User> findByEmail(String email){
+        return userRepository.findByEmail(email);
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/service/UserService.java]: why you return Optional ?

File: src/main/java/com/mulberry/blackmulberry/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void save(User user) throws DuplicateResourceException {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()){
+            throw new DuplicateResourceException("User already exists!");
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/service/UserService.java]: please add dinamic value in your exception message

File: src/main/java/com/mulberry/blackmulberry/service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void save(User user) throws DuplicateResourceException {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()){
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/service/UserService.java]: its okay, but try using Optional's orElseThrow method here. and you can use lambda () ->

File: src/main/java/com/mulberry/blackmulberry/transfer/request/UserRequest.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class UserRequest {
```
Comment (Mushegh128) [inline src/main/java/com/mulberry/blackmulberry/transfer/request/UserRequest.java]: GREATE !!!

File: src/main/java/com/mulberry/blackmulberry/controller/main/OrdersController.java (hunk 5 above)
Comment (hovnaz) [reply src/main/java/com/mulberry/blackmulberry/controller/main/OrdersController.java:22]: Mesrop is responsible for this section․ I already told him

File: src/main/java/com/mulberry/blackmulberry/service/UserService.java (hunk 8 above)
Comment (hovnaz) [reply src/main/java/com/mulberry/blackmulberry/service/UserService.java]: but the userRepository method returns me an Optional, can you send me how to do without the optional?

File: src/main/java/com/mulberry/blackmulberry/service/UserService.java (hunk 9 above)
Comment (hovnaz) [reply src/main/java/com/mulberry/blackmulberry/service/UserService.java]: but it seems to me that the letter is right. the letter indicates that such a user already exists. Please respond if you think there is a more appropriate text

File: src/main/java/com/mulberry/blackmulberry/exception/DuplicateResourceException.java (hunk 7 above)
Comment (hovnaz) [reply src/main/java/com/mulberry/blackmulberry/exception/DuplicateResourceException.java]: can you tell me in more detail what i did wrong?

**Pull request merged**